src/lib/Security/Authentication/RedirectToDashboardAuthenticationSuccessHandler.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Security\Authentication;
+
+use eZ\Publish\Core\MVC\Symfony\Security\Authentication\DefaultAuthenticationSuccessHandler;
+use EzSystems\EzPlatformAdminUiBundle\EzPlatformAdminUiBundle;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Http\HttpUtils;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
+
+class RedirectToDashboardAuthenticationSuccessHandler extends DefaultAuthenticationSuccessHandler
+{
+    use TargetPathTrait;
+
+    /** @var array */
+    private $siteAccessGroups;
+
+    /** @var RouterInterface */
+    private $router;
+
+    /**
+     * @param HttpUtils $httpUtils
+     * @param array $options
+     * @param array $siteAccessGroups
+     * @param RouterInterface $router
+     */
+    public function __construct(
+        HttpUtils $httpUtils,
+        array $options = [],
+        array $siteAccessGroups = [],
+        RouterInterface $router
+    ) {
+        parent::__construct($httpUtils, $options);
+        $this->siteAccessGroups = $siteAccessGroups;
+        $this->router = $router;
+    }
+
+    /**
+     * Builds the target URL according to the defined options.
+     * Overwrites default page after login for admin siteaccess.
+     *
+     * @param Request $request
+     *
+     * @return string
+     */
+    protected function determineTargetUrl(Request $request)
+    {
+        if ($this->isAdmin($request)) {
+            $this->options['default_target_path'] = 'ezplatform.dashboard';
+            $target = $this->getTargetPath($request->getSession(), $this->providerKey);
+            if (null !== $target && 1 === count(explode('/', trim(parse_url($target)['path'], '/')))) {
+                $this->options['always_use_default_target_path'] = true;
+            }
+        }
+
+        return parent::determineTargetUrl($request);
+    }
+
+    /**
+     * Decides if used siteaccess is in admin_group.
+     *
+     * @param Request $request
+     *
+     * @return bool
+     */
+    private function isAdmin(Request $request): bool
+    {
+        $siteAccess = $request->attributes->get('siteaccess');
```

Comment:
Since we are using this routine in many places you can create a helper of some sort.